web/src/stream_topic_history.ts

```diff
@@ -41,12 +45,15 @@ export function all_topics_in_cache(sub) {
     return first_cached_message.id <= sub.first_message_id;
 }
 
-export function is_complete_for_stream_id(stream_id) {
+export function is_complete_for_stream_id(stream_id: number): boolean {
     if (fetched_stream_ids.has(stream_id)) {
         return true;
     }
 
     const sub = sub_store.get(stream_id);
+    if (sub === undefined) {
+        return false;
```

Comment:
I went for this because `all_topics_in_cache` can possibly return false before getting to interacting with `sub`

Reply:
Ah dang now it's failing codecov because of this

Reply:
I'll just do `const in_cache = sub !== undefined && all_topics_in_cache(sub);`